app/Http/Controllers/ApiTokenController.php

```diff
@@ -4,24 +4,36 @@
 
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Auth;
-use App\Models\User;
+use Illuminate\Support\Facades\Gate;
 
 class ApiTokenController extends Controller
 {
     public function showToken(Request $request)
     {
-        return view('showToken', [ 'tokens' => $request->user()->tokens ]);
+        if (!Auth::check()) {
+            return view('welcome');
```

Comment:
Technically, I would also get rid of the welcome view, but this is probably an optimization we can do later.